src/urllib3/poolmanager.py

```diff
@@ -166,25 +184,40 @@ class PoolManager(RequestMethods):
     proxy = None
     proxy_config = None
 
-    def __init__(self, num_pools=10, headers=None, **connection_pool_kw):
+    def __init__(
+        self,
+        num_pools: int = 10,
+        headers: Optional[Mapping[str, str]] = None,
+        **connection_pool_kw: Any,
+    ) -> None:
         super().__init__(headers)
         self.connection_pool_kw = connection_pool_kw
-        self.pools = RecentlyUsedContainer(num_pools, dispose_func=lambda p: p.close())
+        func: Callable[[Any], Any] = lambda p: p.close()
```

Comment:
Can we call this `dispose_func`, also instead of a named lambda should we use a `def` here?